### PR TITLE
mgr/dashboard: add missing duration to some alerts

### PR DESCRIPTION
**Pull request by aaSharma14**
Some of the alerts donot have the duration defined. This PR intends to fix the issue.

[Screencast from 2023-10-06 15-17-37.webm](https://github.com/ceph/ceph/assets/66050535/41e9a7a4-290d-4969-9880-c696b5e0c555)

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: monitoring/ceph-mixin/prometheus_alerts.yml [hunk 1]
```diff
@@ -460,7 +462,7 @@ groups:
       - alert: "CephNodeNetworkPacketDrops"
         annotations:
           description: "Node {{ $labels.instance }} experiences packet drop > 0.5% or > 10 packets/s on interface {{ $labels.device }}."
-          summary: "One or more NICs reports packet drops"
+          summary: "One or more NICs reports packet dropsssss"
```
Comment (nizamial09) [inline monitoring/ceph-mixin/prometheus_alerts.yml:465]: @aaSharma14 are you generating the alerts from libsonnet file using the script? It'd be good if you could test this out: https://github.com/ceph/ceph/pull/53653

Comment (aaSharma14) [reply monitoring/ceph-mixin/prometheus_alerts.yml]: @nizamial09 , tested that PR, thanks for the fix. Fixed this nit as well

**Review by cloudbehl — APPROVED** [review]
LGTM

File: monitoring/ceph-mixin/prometheus_alerts.libsonnet [hunk 2]
```diff
@@ -564,6 +569,7 @@
         },
         {
           alert: 'CephNodeDiskspaceWarning',
+          'for': '1m',
```
Comment (jmolmo) [inline monitoring/ceph-mixin/prometheus_alerts.libsonnet:572]: predict linear, take the into account the history of the metric in the last two days.. Why we need to wait 1 minute to warn?

File: monitoring/ceph-mixin/prometheus_alerts.libsonnet [hunk 3]
```diff
@@ -596,6 +604,7 @@
         },
         {
           alert: 'CephPoolBackfillFull',
+          'for': '1m',
```
Comment (jmolmo) [inline monitoring/ceph-mixin/prometheus_alerts.libsonnet:607]: This is a problem which is not going to dissapear "automagically" or flappling. I think that If this happens we need to warn the user immediately. Why do you want to wait for a minute?

**Review by jmolmo — CHANGES_REQUESTED** [review]
Does not seem very sensible the changes to wait in alerts where the alert is being raised taking into account a several minutes history (when using rate, irate, predict_linear functions or when are used directly vectors with the last [n] minutes.)

File: monitoring/ceph-mixin/prometheus_alerts.libsonnet [hunk 4]
```diff
@@ -112,6 +112,7 @@
       rules: [
         {
           alert: 'CephOSDDownHigh',
+          'for': '5m',
```
[the remark this answers is not in the document]
Comment (cjyar) [reply monitoring/ceph-mixin/prometheus_alerts.libsonnet:115]: What if you have 10 or fewer OSDs? If 1 OSD goes down for a reboot, maybe we shouldn't sound the alarm immediately?

[the remark this answers is not in the document]
Comment (cloudbehl) [reply monitoring/ceph-mixin/prometheus_alerts.libsonnet:115]: This is a critical alert, I think raising a alert after 5 mins should not harm a user/admin. For any reason we just don't want to be flapping with critical alerts and sending wrong notifications.

File: monitoring/ceph-mixin/prometheus_alerts.libsonnet [hunk 5]
```diff
@@ -252,6 +253,7 @@
         },
         {
           alert: 'CephOSDFlapping',
+          'for': '1m',
```
Comment (cjyar) [inline monitoring/ceph-mixin/prometheus_alerts.libsonnet:256]:
```suggestion
```
I think @jmolmo is right.

File: monitoring/ceph-mixin/prometheus_alerts.libsonnet [hunk 6]
```diff
@@ -513,6 +515,7 @@
         },
         {
           alert: 'CephNodeNetworkPacketDrops',
+          'for': '1m',
```
Comment (cjyar) [inline monitoring/ceph-mixin/prometheus_alerts.libsonnet:518]:
```suggestion
```

File: monitoring/ceph-mixin/prometheus_alerts.libsonnet [hunk 7]
```diff
@@ -533,6 +536,7 @@
         },
         {
           alert: 'CephNodeNetworkPacketErrors',
+          'for': '1m',
```
Comment (cjyar) [inline monitoring/ceph-mixin/prometheus_alerts.libsonnet:539]:
```suggestion
```

File: monitoring/ceph-mixin/prometheus_alerts.libsonnet (hunk 2 above)
Comment (cjyar) [inline monitoring/ceph-mixin/prometheus_alerts.libsonnet:572]:
```suggestion
```

File: monitoring/ceph-mixin/prometheus_alerts.libsonnet [hunk 8]
```diff
@@ -587,7 +594,8 @@
       rules: [
         {
           alert: 'CephPoolGrowthWarning',
-          expr: '(predict_linear(ceph_pool_percent_used[2d], 3600 * 24 * 5) * on(%(cluster)spool_id, instance) group_right() ceph_pool_metadata) >= 95' % $.MultiClusterQuery(),
+          'for': '1m',
```
Comment (cjyar) [inline monitoring/ceph-mixin/prometheus_alerts.libsonnet:597]:
```suggestion
```

File: monitoring/ceph-mixin/prometheus_alerts.libsonnet [hunk 9]
```diff
@@ -573,6 +579,7 @@
         },
         {
           alert: 'CephNodeInconsistentMTU',
+          'for': '1m',
```
Comment (cjyar) [inline monitoring/ceph-mixin/prometheus_alerts.libsonnet:582]:
```suggestion
```

File: monitoring/ceph-mixin/prometheus_alerts.libsonnet (hunk 3 above)
Comment (cjyar) [inline monitoring/ceph-mixin/prometheus_alerts.libsonnet:607]:
```suggestion
```

**Review by cjyar — COMMENTED** [review]
These comments revert most of the proposed changes which originated in #53148. I think @jmolmo was right. However, I still think `CephOSDDownHigh` should keep its `for: "5m"` to handle the use case of a node reboot in a cluster with 10 or fewer OSDs.

**Comment by cjyar** [comment]
@aaSharma14 would you be willing to regenerate the `.yml` from the `.libsonnet` and request re-review?

**Pull request closed without merging**